src/fondant/compiler.py

```diff
@@ -263,28 +264,32 @@ def kfp_pipeline():
                 logger.info(f"Compiling service for {component_name}")
 
                 # convert ComponentOp to Kubeflow component
-                kubeflow_component_op = self.kfp.components.load_component(
+                kubeflow_component_op = self.kfp.components.load_component_from_text(
                     text=component_op.component_spec.kubeflow_specification.to_string(),
                 )
 
+                # # Set image pull policy to always
                 # Execute the Kubeflow component and pass in the output manifest path from
                 # the previous component.
                 component_args = component_op.arguments
 
-                component_task = kubeflow_component_op(
-                    input_manifest_path=manifest_path,
-                    metadata=metadata.to_json(),
-                    **component_args,
-                )
-                # Set optional configurations
-                component_task = self._set_configuration(
-                    component_task,
-                    component_op,
-                )
-
-                # Set image pull policy to always
-                component_task.container.set_image_pull_policy("Always")
```

Comment:
might be nice to keep the image pull policy (unless it's another function in v2 in which case we can tackle it separately or create a task for it)

Reply:
added as a task: https://github.com/ml6team/fondant/issues/393